packages/beagle-react/src/components/BeagleDynamicLists/DynamicListCore/styled.ts

```diff
@@ -15,24 +15,29 @@
 */
 
 import styled from 'styled-components'
-import { Direction } from 'common/models'
+import { Direction, ListType } from 'common/models'
 
-interface StyledListViewInterface {
+interface StyledDynamicViewsInterface {
   direction: Direction,
   useParentScroll?: boolean,
   isScrollIndicatorVisible?: boolean,
+  listType?: ListType,
+  numColumns?: number,
 }
 
-export const StyledListView = styled.div<StyledListViewInterface>`
-  display: flex;
+export const StyledDynamicViewsInterface = styled.div<StyledDynamicViewsInterface>`
+  display: ${({ listType }) => listType === 'GRID' ? 'grid' : 'flex'};
+  grid-template-columns: ${({ listType, numColumns }) => 
+    (listType === 'GRID' && numColumns) && `repeat(${numColumns}, auto)`};
+  justify-content: ${({ listType })=> listType === 'GRID' && 'space-evenly'};
   flex-direction: ${({ direction }) => direction === 'VERTICAL' ? 'column' : 'row'};
   overflow: ${({ useParentScroll }) => useParentScroll ? 'inherit' : 'auto'};
   width: ${({ direction }) => direction === 'HORIZONTAL' ? '100%' : 'auto'};
   height: ${({ direction }) => direction === 'VERTICAL' ? '100%' : 'auto'};
```

Comment:
Based on the suggestion of using a String Enum, instead of two strings as type, I suggest changing here to this approach (also using this opportunity to do the same change on direction):

```suggestion
export const StyledDynamicViewsInterface = styled.div<StyledDynamicViewsInterface>`
  display: ${({ listType }) => listType === ListType.Grid ? 'grid' : 'flex'};
  grid-template-columns: ${({ listType, numColumns }) => 
    (listType === ListType.Grid && numColumns) && `repeat(${numColumns}, auto)`};
  justify-content: ${({ listType })=> listType === ListType.Grid && 'space-evenly'};
  flex-direction: ${({ direction }) => direction === ListDirection.Vertical ? 'column' : 'row'};
  overflow: ${({ useParentScroll }) => useParentScroll ? 'inherit' : 'auto'};
  width: ${({ direction }) => direction === ListDirection.Horizontal ? '100%' : 'auto'};
  height: ${({ direction }) => direction === ListDirection.Vertical ? '100%' : 'auto'};
```